app/validators/RegisterValidator.ts

```diff
@@ -0,0 +1,34 @@
+import { schema, CustomMessages, rules } from '@ioc:Adonis/Core/Validator'
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+
+export default class RegisterValidator {
+  constructor(protected ctx: HttpContextContract) {}
+
+  public schema = schema.create({
+    email: schema.string({ trim: true }, [
+      rules.email(),
+      rules.unique({ table: 'user', column: 'email', caseInsensitive: true }),
+    ]),
+    password: schema.string({ trim: true }, [
+      rules.regex(/^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,16}$/),
+    ]),
+    mobile: schema.string({}, [
+      rules.unique({ table: 'profile', column: 'mobile' }),
+      rules.mobile(),
+      rules.maxLength(10),
+      rules.minLength(10),
+    ]),
+    gender: schema.enum(Object.values(Gender)),
+    dob: schema.date(),
```

Comment:
```suggestion
    dob: schema.date({ format: 'yyyy-MM-dd' }),
```

pass format for proper validation